seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/SpringCloudAlibabaConfiguration.java

```diff
@@ -0,0 +1,73 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.spring.boot.autoconfigure.properties;
+
+import io.seata.spring.boot.autoconfigure.StarterConstants;
+import org.springframework.beans.BeansException;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+/**
+ * The type Spring cloud alibaba configuration.
+ *
+ * @author jimin.jm @alibaba-inc.com
+ * @date 2019 /11/04
+ */
+@ConfigurationProperties(prefix = StarterConstants.SEATA_SPRING_CLOUD_ALIBABA_PREFIX)
+public class SpringCloudAlibabaConfiguration implements ApplicationContextAware {
+    private static final String SPRING_APPLICATION_NAME_KEY = "spring.application.name";
+    private static final String DEFAULT_SPRING_CLOUD_SERVICE_GROUP_POSTFIX = "-seata-service-group";
+    private String applicationId;
+    private String txServiceGroup;
+    private ApplicationContext applicationContext;
+
+    /**
+     * Gets application id.
+     *
+     * @return the application id
+     */
+    public String getApplicationId() {
+        applicationId = applicationContext.getEnvironment().getProperty(SPRING_APPLICATION_NAME_KEY);
+        return applicationId;
+    }
+
+    /**
+     * Gets tx service group.
+     *
+     * @return the tx service group
+     */
+    public String getTxServiceGroup() {
+        if (null == txServiceGroup) {
+            txServiceGroup = applicationId + DEFAULT_SPRING_CLOUD_SERVICE_GROUP_POSTFIX;
```

Comment:
if user not configuration spring.application.name then applicationId is null?

Reply:
> if user not configuration spring.application.name then applicationId is null?

If user does not configure this item,the default value is "application".
![image](https://user-images.githubusercontent.com/24457941/68279551-9e65c280-00ae-11ea-8875-af40e7e7b735.png)

Reply:
![image](https://user-images.githubusercontent.com/26616196/68304665-37adcc80-00e1-11ea-8478-cda9016ee22d.png)
If user does not configuration, the applicationId is null, the txServiceGroup is null-seata-service-group?

Reply:
> ![image](https://user-images.githubusercontent.com/26616196/68304665-37adcc80-00e1-11ea-8478-cda9016ee22d.png)
> If user does not configuration, the applicationId is null, the txServiceGroup is null-seata-service-group?

Yes,here could be null.

Reply:
We can use "application" replace null.

Reply:
I remember long long ago, the `spring.application.name` is must not be null.